core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -95,7 +106,7 @@ protected void inboundTransportError(Status errorStatus) {
     }
     // For transport errors we immediately report status to the application layer
     // and do not wait for additional payloads.
-    transportReportStatus(errorStatus, false, new Metadata());
+    transportReportStatus(errorStatus, false, metadata);
   }
 
   /**
```

Comment:
Since I can't comment on lines you didn't add:  Until the new compression change is in, you should probably also change line 132 to include the headers.